kale/utils/metadata_utils.py

```diff
@@ -0,0 +1,103 @@
+import re
+
+from kale.utils.pod_utils import is_workspace_dir
+
+METADATA_REQUIRED_KEYS = [
+    'experiment_name',
+    'pipeline_name',
+]
+kale_step_name_regex = r'^[a-z0-9]([-a-z0-9]*[a-z0-9])?$'
+kale_name_msg = ("must consist of lower case alphanumeric characters "
+                 "or '-', and must start and end with an alphanumeric"
+                 " character.")
+k8s_valid_name_regex = r'^[\.\-a-z0-9]+$'
+k8s_name_msg = ("must consist of lower case alphanumeric characters, "
+                "'-' or '.'")
+volume_types = ['pv', 'pvc', 'new_pvc', 'clone']
+volume_required_fields = ['name', 'annotations', 'size', 'type', 'mount_point']
+
+
+def validate_metadata(metadata):
+    """Validate the Notebook's metadata and update it inplace when needed.
+
+    Args:
+        metadata (dict): metadata annotated by Kale. Refer to DEFAULT_METADATA
+    """
+    # check for required fields
+    for required in METADATA_REQUIRED_KEYS:
+        if required not in metadata:
+            raise ValueError("Key %s not found. Add this field either on "
+                             "the notebook metadata or as an override" %
+                             required)
+
+    if not re.match(kale_step_name_regex, metadata['pipeline_name']):
+        raise ValueError("Pipeline name  %s" % kale_name_msg)
+
+    volumes = metadata.get('volumes', [])
+    if volumes or isinstance(volumes, list):
```

Comment:
It should be `and` here, instead of `or`.

Reply:
Actually we should remove `volumes` and have just:
```python
if isinstance(volume, list):
```
Otherwise the condition would never pass with an empty list